planner/core/logical_plans.go

```diff
@@ -179,6 +181,77 @@ func (p *LogicalJoin) Shallow() *LogicalJoin {
 	return join.Init(p.ctx, p.blockOffset)
 }
 
+// ExtractFD implements the interface LogicalPlan.
+func (p *LogicalJoin) ExtractFD() *fd.FDSet {
+	switch p.JoinType {
```

Comment:
`logical_plans.go` is large enough, how about creating a dedicated and separate file for FD in this folder?

Reply:
Let us split later. We need to resolve conflicts with the feature branch.